controllers/vspherecluster_reconciler.go

```diff
@@ -351,6 +353,23 @@ func (r clusterReconciler) reconcileDeploymentZones(ctx *context.ClusterContext)
 	return true, nil
 }
 
+func shouldIncludeZone(zone infrav1.VSphereDeploymentZone, cluster *infrav1.VSphereCluster) bool {
+	// expects cluster to be annotated
+	// vsphere.infrastructure.cluster.x-k8s.io/adopt-deploymentzone: "dz1, dz2"
+	// if there are there deploymentzones available
+	// dz1, dz2 and dz3
+	// this annotation will allow cluster to include dz1 and dz2
+	deploymentZones, annotationExists := cluster.GetAnnotations()["vsphere.infrastructure.cluster.x-k8s.io/adopt-deploymentzone"]
```

Comment:
I'd suggest to move that annotation key into a constant and maybe add some unit tests under `vsphere_cluster_reconciler_test.go`

Reply:
Ack